### PR TITLE
chore: standardize samples directory - delete unneeded dependencies

### LINKED ISSUE #790: standardize samples directory layout
We have an odd mix of sample locations:

- `docs/snippets.py` -- the original location, not very copy-pasteable / browsable
- lots of samples in `samples/*.py`, but the test drivers are in `samples/tests/`
- `samples/snippets/*.py`
- `samples/geography/*.py`

To align with the infrastructure in https://github.com/googleapis/synthtool/tree/master/synthtool/gcp/templates/python_samples, the goal is to have:

- `samples/snippets/` -- these samples just require the `google-cloud-bigquery` library 
- `samples/geography/` -- require extra dependencies for working with geography data
- `sample/pandas/` -- require extra dependencies for working with pandas dataframes
- more??? -- possibly a directory for the user-based authentication samples

To avoid breaking cloud.google.com docs, this needs to be done in several steps:

1. Copy samples to correct location
2. Update docs (Googler must do this)
3. Delete duplicate samples in old location

Towards: https://github.com/googleapis/python-bigquery/issues/366 (use templated noxfile) and ~https://github.com/googleapis/python-bigquery/issues/765 (blacken samples using templated noxfile(s))~

### PR DESCRIPTION
**Pull request by kiraksi**
This PR:
- is a follow up to https://github.com/googleapis/python-bigquery/pull/1727
- should not merge unless internal cl to update documentation is submitted
- removes unneeded dependencies from samples directories's requirements files
- adds conftest fixture to speed things up slightly

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes #790 🦕

### CODE REVIEW

**Review by tswast — COMMENTED** [review]
Lovely! I'll wait to approve until it's ready and the internal CL is submitted.

**Comment by kiraksi** [comment]
> Lovely! I'll wait to approve until it's ready and the internal CL is submitted.

Internal CL has been submitted but I assume we need to wait until after the freeze period to merge!

**Review by chalmerlowe — APPROVED** [review]
LGTM

**Pull request merged**
